packages/amplify-cli/src/__tests__/extensions/amplify-helpers/get-all-category-pluginInfos.test.ts

```diff
@@ -1,10 +1,10 @@
 import { constructContext } from '../../../context-manager';
 
-import { PluginCollection } from 'amplify-cli-core';
+import { $TSContext, PluginCollection } from 'amplify-cli-core';
 import { constructMockPluginPlatform } from './mock-plugin-platform';
 
 import { CLIInput as CommandLineInput } from '../../../domain/command-input';
-import { getAllCategoryPluginInfo } from '../../../extensions/amplify-helpers/get-all-category-pluginInfos';
+import { getAllCategoryPluginInfo } from 'amplify-cli-core/lib/extensions/get-all-category-pluginInfos';
```

Comment:
is import from `lib` legal in tests ? (in other files too)

Reply:
A few of Toolkit's imports modify or rely on the application's state. Meaning, if you don't import them in the right order, the application will crash. So, exporting all of the files in a barrel file in `amplify-cli-core` could cause undesired behavior. I did the `lib/` import pattern to avoid this issue. However, since it is the minority of Toolkit's dependencies that behave in this unexpected way, what we can do is export most of these values in a barrel file (the current pattern used across `amplify-cli-core`), and handle the special cases differently as we come across them.